### PR TITLE
Add AWS SDK and Soto examples

### PR DESCRIPTION
**Pull request by sebsto**

### CODE REVIEW

File: Examples/AWSSDK/Sources/main.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import AWSLambdaRuntime
+import AWSS3
+
+func handler(event: APIGatewayV2Request, context: LambdaContext) async throws -> APIGatewayV2Response {
+
+    var response: APIGatewayV2Response
+    do {
+        // read the list of buckets
+        context.logger.debug("Reading list of buckets")
+        let client = try await S3Client()
```
Comment (fabianfett) [inline Examples/AWSSDK/Sources/main.swift:25]: recreating the client over and over looks wasteful. create the client outside the scope of this closure.

File: Examples/Soto/Sources/main.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import AWSLambdaEvents
+import AWSLambdaRuntime
+import SotoS3
+
+let client = AWSClient()
```
Comment (fabianfett) [inline Examples/Soto/Sources/main.swift:19]: please pass the ELG here.

File: Examples/Soto/template.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+#
+# Copyright (c) 2024, the VSCode AWS Lambda Swift extension project authors.
+# Licensed under Apache License v2.0.
+#
+# See LICENSE.txt for license information
+# See CONTRIBUTORS.txt for the list of VSCode AWS Lambda Swift project authors
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+#===----------------------------------------------------------------------===//
```
Comment (fabianfett) [inline Examples/Soto/template.yaml:14]: this header will not pass the license check.

**Review by fabianfett — CHANGES_REQUESTED** [review]
Love it! First round of reviews.

File: Examples/Soto/Sources/main.swift (hunk 2 above)
Comment (sebsto) [reply Examples/Soto/Sources/main.swift:19]: ELG ?

File: Examples/APIGateway/README.md [hunk 4]
```diff
@@ -40,9 +40,9 @@ To actually deploy your Lambda function and create the infrastructure, type the
 
 ```bash
 sam deploy \
-----resolve-s3 \
+--resolve-s3 \
 --template-file template.yaml \
---stack-name MyLambda \
+--stack-name APIGAtewayLambda \
```
[the remark this answers is not in the document]
Comment (sebsto) [reply Examples/APIGateway/README.md:45]: mmmh weird, I already changed that

File: Examples/AWSSDK/Sources/main.swift (hunk 1 above)
Comment (sebsto) [reply Examples/AWSSDK/Sources/main.swift:25]: OK, fixed with a @preconcurrency

File: Examples/AWSSDK/Package.swift [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+// needed for CI to test the local version of the library
+import class Foundation.ProcessInfo
+import struct Foundation.URL
+
+#if os(macOS)
+let platforms: [PackageDescription.SupportedPlatform]? = [.macOS(.v15)]
+#else
+let platforms: [PackageDescription.SupportedPlatform]? = nil
+#endif
```
[the remark this answers is not in the document]
Comment (sebsto) [reply Examples/AWSSDK/Package.swift]: it works with docker swift-amazonlinux2 without that #if - I removed it

File: Examples/Soto/Sources/main.swift (hunk 2 above)
Comment (sebsto) [reply Examples/Soto/Sources/main.swift:19]: Do you have an Example, I can't find a way to access the underlying EventLoop on Soto 7. I think everything is hidden now as implementation details

Comment (fabianfett) [reply Examples/Soto/Sources/main.swift:19]: yeah. true. both should use the same thanks to `NIOSingletons.posixEventLoogProup`. Consider this resolved.

File: Examples/AWSSDK/Sources/main.swift [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+//
+//===----------------------------------------------------------------------===//
+
+import AWSLambdaEvents
+import AWSLambdaRuntime
+@preconcurrency import AWSS3
+
+let client = try await S3Client()
+
+let runtime = LambdaRuntime.init {
```
Comment (fabianfett) [inline Examples/AWSSDK/Sources/main.swift:21]:
```suggestion
let runtime = LambdaRuntime {
```

Comment (sebsto) [reply Examples/AWSSDK/Sources/main.swift]: Done

**Comment by sebsto** [comment]
Waiting to land the CI change (https://github.com/swift-server/swift-aws-lambda-runtime/pull/397) before merging this one.

I will need to add `AWSSDK` and `SotoS3` to the `examples_matrix.yml` file

**Review by sebsto — COMMENTED** [review]
all changes are implemented

File: Examples/S3_AWSSDK/Package.swift [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+let package = Package(
+    name: "AWSSDKExample",
+    platforms: platforms,
+    products: [
+        .executable(name: "AWSSDKExample", targets: ["AWSSDKExample"])
+    ],
+    dependencies: [
+        // dependency on swift-aws-lambda-runtime is added dynamically below
+        // .package(url: "https://github.com/swift-server/swift-aws-lambda-runtime.git", branch: "main")
+        .package(url: "https://github.com/swift-server/swift-aws-lambda-events", branch: "main"),
```
Comment (fabianfett) [inline Examples/S3_AWSSDK/Package.swift:24]: this looks wrong

File: Examples/S3_AWSSDK/template.yaml [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+Description: SAM Template for AWS SDK Example
+
+Resources:
+  # Lambda function
+  AWSSDKExample:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: .build/plugins/AWSLambdaPackager/outputs/AWSLambdaPackager/AWSSDKExample/AWSSDKExample.zip
+      Timeout: 60
+      Handler: swift.bootstrap
```
Comment (fabianfett) [inline Examples/S3_AWSSDK/template.yaml:12]: we may want to document that swift ignores this property.

File: Examples/S3_AWSSDK/template.yaml [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+      Runtime: provided.al2
+      MemorySize: 512
+      Architectures:
+        - arm64
+      Environment:
+        Variables:
+          # by default, AWS Lambda runtime produces no log
+          # use `LOG_LEVEL: debug` for for lifecycle and event handling information
+          # use `LOG_LEVEL: trace` for detailed input event information
+          LOG_LEVEL: trace
```
Comment (fabianfett) [inline Examples/S3_AWSSDK/template.yaml:22]: do we really want trace here? I think thats a bad example.

File: Examples/S3_Soto/Package.swift [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+    platforms: platforms,
+    products: [
+        .executable(name: "SotoLambdaExample", targets: ["SotoExample"])
+    ],
+    dependencies: [
+        .package(url: "https://github.com/soto-project/soto.git", from: "7.0.0"),
+
+        // dependency on swift-aws-lambda-runtime is added dynamically below
+        // .package(url: "https://github.com/swift-server/swift-aws-lambda-runtime.git", branch: "main")
+        .package(url: "https://github.com/swift-server/swift-aws-lambda-events", branch: "main"),
```
Comment (fabianfett) [inline Examples/S3_Soto/Package.swift:26]: that looks wrong.

File: Examples/S3_Soto/Package.swift [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+            dependencies: [
+                .product(name: "SotoS3", package: "soto"),
+                .product(name: "AWSLambdaRuntime", package: "swift-aws-lambda-runtime"),
+                .product(name: "AWSLambdaEvents", package: "swift-aws-lambda-events"),
+            ]
+        )
+    ]
+)
+
+if let localDepsPath = ProcessInfo.processInfo.environment["LAMBDA_USE_LOCAL_DEPS"],
```
Comment (fabianfett) [inline Examples/S3_Soto/Package.swift:40]: We could use `Context.environment` instead of `ProcessInfo.processInfo.environment` here.

**Review by fabianfett — COMMENTED** [review]
I think we must fix the dependencies. Everything else is optional.

**Comment by fabianfett** [comment]
please enable CI for the new examples :)

**Comment by sebsto** [comment]
> please enable CI for the new examples :)

I'm on it

**Review by fabianfett — APPROVED** [review]
We can fix everything left in follow ups, if we want to.

File: Examples/S3_AWSSDK/template.yaml (hunk 9 above)
Comment (sebsto) [reply Examples/S3_AWSSDK/template.yaml]: OK, I'll add a note to verify that the TRACE level has been kept in the new runtime

**Pull request merged**